packages/@glimmer-workspace/integration-tests/test/debug-render-tree-test.ts

```diff
@@ -74,6 +77,88 @@ class DebugRenderTreeTest extends RenderTest {
 
   declare delegate: DebugRenderTreeDelegate;
 
+  @test 'strict-mode components'() {
+    const state = trackedObj({ showSecond: false });
+
+    const HelloWorld = defComponent('{{@arg}}');
+    const Root = defComponent(
+      `<HelloWorld @arg="first"/>{{#if state.showSecond}}<HelloWorld @arg="second"/>{{/if}}`,
+      { scope: { HelloWorld, state }, emit: { moduleName: 'root.hbs' } }
+    );
+
+    this.renderComponent(Root);
+
+    this.assertRenderTree([
+      {
+        type: 'component',
+        name: '{ROOT}',
+        args: { positional: [], named: {} },
+        instance: null,
+        template: 'root.hbs',
+        bounds: this.elementBounds(this.delegate.getInitialElement()),
+        children: [
+          {
+            type: 'component',
+            name: 'HelloWorld',
+            args: { positional: [], named: { arg: 'first' } },
+            instance: null,
+            template: '(unknown template module)',
```

Comment:
would be good to drop this, I think.
it's not helpful to have `(unknown template module)` for every strict-mode component -- they .. have no template?, yea? they aren't separate modules anymore.

Reply:
I plan to do some follow-up PRs that finish the job of making the render tree reflect what the user typed. For now, I tried to leave things alone if they weren't directly on point with the specific changes here.

Specifically regarding `template`, I think we should consider whether we can replace it with module information. There's more to say about that, and I'd want to check with @ef4 about (a) whether it's realistic, and (b) what the right information to include would be.

In the short-term, though, the `template` field is increasingly likely to say `(unknown template module)` (and that's reliably true in the GJS format). We therefore never want to display it to users as-is and I agree it should be removed in a follow-up PR.